notcurses: init at 2.1.0

andir opened the pull request:
I'm forwarding this patch that I received via email:

```
commit 16b7254d1d1fce8f1d4e439ac3cbd491d1dfa622
Author: William Casarin <jb55@jb55.com>
Date:   Tue Dec 15 07:01:09 2020 -0800

    notcurses: init at 2.1.0

    This is a neat ncurses alternative. See the demo here:

    https://www.youtube.com/watch?v=cYhZ7myXyyg

    Here's something interesting to package after this:

    https://github.com/dankamongmen/growlight

    Message-Id: <20201215150109.30656-1-jb55@jb55.com>
```

You can find the submission in the [archive](https://lists.sr.ht/~andir/nixpkgs-dev/%3C20201215150109.30656-1-jb55%40jb55.com%3E).

xaverdh commented:
Really interesting project!
Should this be split into a `dev` output for the library part and `bin` for the binaries maybe?

xaverdh commented:
I built and tried it out, all binaries appear to run fine.

andir commented:
On 08:58 17.12.20, xaverdh wrote:
> Really interesting project!
> Should this be split into a `dev` output for the library part and `bin` for the binaries maybe?

We could probably do that even thought I do not see much of a benefit
(yet). It is mostly being used as library (I suppose?) and the binaries
are mostly examples? Either way works for me.

xaverdh commented:
> On 08:58 17.12.20, xaverdh wrote: Really interesting project! Should this be split into a `dev` output for the library part and `bin` for the binaries maybe?
> We could probably do that even thought I do not see much of a benefit (yet). It is mostly being used as library (I suppose?) and the binaries are mostly examples? Either way works for me.

I think the `notcurses-view` might be useful on its own, but yes this can wait until the library / project is more widely used.

dankamongmen commented:
> I built and tried it out, all binaries appear to run fine.
> 
> **edit**: But no audio with `notcurses-view` (might be intentional / just not implemented though?).

no audio is intended, though it could certainly be added if people wanted it

dankamongmen commented:
One project that is packaged in Debian and Fedora, but I've not yet packaged for NixOS, is [growlight](https://github.com/dankamongmen/growlight). Another is [snd](https://sourceforge.net/p/snd/svn1/HEAD/tree/). I'd be interested in producing a NixOS recipe for Growlight once Notcurses went in.

mcbeth reviewed (approved):
This worked on my ubuntu+nixpkgs and my NixOS machine.  It probably should go into the maintainer's thought queue to support the QR code stuff (I'll never need it, but it is an advertised part of the code base) in the future.

xaverdh commented:
> This worked on my ubuntu+nixpkgs and my NixOS machine. It probably should go into the maintainer's thought queue to support the QR code stuff (I'll never need it, but it is an advertised part of the code base) in the future.

This requires packaging libqrencode, right? I actually tried to do this locally but it appears that `notcurses` expects to find the library files relative to its own source directory..

jb55 commented:
@mcbeth@github.com writes:
> This worked on my ubuntu+nixpkgs and my NixOS machine. It probably
> should go into the maintainer's thought queue to support the QR code
> stuff (I'll never need it, but it is an advertised part of the code
> base) in the future.

I just wanted to have something simple for now without supporting every
possible build option. I wasn't sure what the QR stuff was about.

@xaverdh@github.com writes:
> Really interesting project!
> Should this be split into a `dev` output for the library part and
> `bin` for the binaries maybe?

I think `out` and `dev` makes sense, since it looks like bin requires
some additional data from `share`. I'll push this in v2.

Cheers,
Will

jb55 committed:
notcurses: init at 2.1.0

This is a neat ncurses alternative. See the demo here:

https://www.youtube.com/watch?v=cYhZ7myXyyg

Here's something interesting to package after this:

https://github.com/dankamongmen/growlight

Message-Id: <20201218142929.9912-1-jb55@jb55.com>

dankamongmen commented:
> > This worked on my ubuntu+nixpkgs and my NixOS machine. It probably should go into the maintainer's thought queue to support the QR code stuff (I'll never need it, but it is an advertised part of the code base) in the future.
> 
> This requires packaging libqrencode, right? I actually tried to do this locally but it appears that `notcurses` expects to find the library files relative to its own source directory..

no, that shouldn't be the case. on debian, i build against `libqrcodegen-dev`. on fedora, `libqrcodegen-devel`, etc. either way, the qrcodegen stuff is of pretty esoteric appeal.

mcbeth commented:
[default.nix.txt](https://github.com/NixOS/nixpkgs/files/5717512/default.nix.txt)

I've attached a simple/sample qrcodegen default.nix that is enough to enable the support in notcurses.  There's 6 other languages in that repo that I'm ignoring right now...

andir commented:
Added the dev output. Any more remarks besides the qrcodegen?

@mcbeth mind opening a PR for qrcodegen?

dankamongmen commented:
so is this going in? (sorry, i'm a bit ignorant of the nixos workflow)

jb55 commented:
> so is this going in? (sorry, i'm a bit ignorant of the nixos workflow)

It goes in once someone who has commit access decides to commit it :P
